src/collapse/mod.rs

```diff
@@ -15,12 +15,28 @@ pub mod guess;
 ///   [crate-level documentation]: ../../index.html
 pub mod perf;
 
+///////////////////////////////////////////////////////////////////////////////
+
+use std::cmp;
 use std::fs::File;
 use std::io::{self, Write};
+use std::mem;
 use std::path::Path;
+use std::sync::Arc;
+
+use chashmap::CHashMap;
+use fnv::FnvHashMap;
 
+///////////////////////////////////////////////////////////////////////////////
+
+const CAPACITY_INPUT_BUFFER: usize = 1024 * 1024 * 1024;
+const CAPACITY_LINE_BUFFER: usize = 1024;
+const CAPACITY_MULTI_THREADED_HASHMAP: usize = 64;
+const CAPACITY_SINGLE_THREADED_HASHMAP: usize = 512;
```

Comment:
Why is `MULTI` and `SINGLE` different?

Reply:
The master branch is using 512 for its FNV hashmap. The concurrent hashmap is a bit of a different beast; so I just started out with it's default (which is 64 if you look at chashmap's codebase). The intent was to change that 64 around to see if it makes much difference. As I recall, it doesn't really make a difference; so I just left it at the library's default. We can use one number for both if you want. Just let me know.

Reply:
By the way, I only have a couple of minutes now; so I'm just leaving short notes for things I think I can answer right away, but probably won't get to the bulk of comments until later.

Reply:
I think I'd probably just use a single constant for both; it's not clear to me that there'd ever be a reason to use different values for them.